src/Components/Search/NewSearch/utils/formatOptions.ts

```diff
@@ -0,0 +1,37 @@
+import { getLabel } from "./getLabel"
+import { SuggionItemOptionProps } from "Components/Search/NewSearch/SuggestionItem/NewSuggestionItem"
+
+export interface SearchNodeOption {
+  __typename: string
+  displayLabel?: string
+  displayType?: string
+  imageUrl?: string
+  statuses?: {
+    artworks?: boolean
+    auctionLots?: boolean
+  }
+  href?: string
+}
+
+export const formatOptions = (
+  options: SearchNodeOption[]
+): SuggionItemOptionProps[] => {
+  return options.map((option, index) => {
+    return {
+      text: option.displayLabel!,
+      value: option.displayLabel!,
+      subtitle:
+        getLabel({
+          displayType: option.displayType ?? "",
+          typename: option.__typename,
+        }) ?? "",
+      imageUrl: option.imageUrl!,
+      showArtworksButton: !!option.statuses?.artworks,
+      showAuctionResultsButton: !!option.statuses?.auctionLots,
+      href: option.href!,
+      typename: option.__typename,
+      item_number: index,
+      item_type: option.displayType!,
+    }
+  })
```

Comment:
Not sure why we would do something like this; build fragment containers instead.

Reply:
(This kind of thing makes refactors extremely difficult)

Reply:
@dzucconi - mind elaborating a bit on what this would ideally look like? Like just move this logic into a component and tie the props to the relay generated type?

Reply:
To be clear: there should be a fragment container for the search result items, and those components can handle this formatting themselves.

Reply:
Yeah, makes sense, I will refactor this